common/persistence/dataManagerInterfaces.go

```diff
@@ -1372,6 +1371,7 @@ type (
 		VisibilityArchivalStatus types.ArchivalStatus
 		VisibilityArchivalURI    string
 		BadBinaries              types.BadBinaries
+		IsolationGroups          *types.IsolationGroupConfiguration
```

Comment:
I see inconsistencies of this field in different types. In some types, you use pointer, and in some types you use value and in some types you use the array type `[]types.IsolationGroupPartition`. 
It creates lots of confusion to people. I'm not sure when we should use which type.

Reply:
yeah, I think that's completely fair criticism, I've just been a bit sloppy, let me remove the pointer as it's a map and I don't really need to include it